app/lib/vsphere_os_identifiers.rb

```diff
@@ -21,7 +21,17 @@ def self.where(selectors = {})
 
     selectors.each do |selector, value|
       next unless selectors.key?(selector)
-      found.select! { |os| os.public_send(selector) && os.public_send(selector) == value }
+      found.select! do |os|
+        test = os.public_send(selector)
```

Comment:
Do you mind extracting this to a dedicated method?
Or how about

```ruby
found.select! { |os| os.public_send(selector) && Array.new(os.public_send(selector)).flatten.include?(value) 
```

?

Reply:
The code I pushed was first built to match on regexp as well, but I redid it with just arrays instead since that didn't require me to post-process the parsed YAML. So that switch originally had a few more cases, but this seems a much saner solution.